### PR TITLE
Nft improvement

### PR DESCRIPTION
**Pull request by superokid**
- Fix NFT detail attributes should show if visiting the page from homepage
- Fix NFT detail with different spec show attributes from extra
- Should hide button loadmore if less than the limit per page
- Add loading bar when fetch attributes

Screenshot 2024-04-26 at 17 30 52

### CODE REVIEW

File: packages/frontend/src/redux/slices/nft/index.js
Comment (SteveKok) [inline packages/frontend/src/redux/slices/nft/index.js:1]:
```typescript
for (let i = 0; i < arr.length; i++) {
      const item = arr[i];
      if (seen[item.token_id] !== undefined) {
          arr[seen[item.token_id]] = item;
          seenCount++;
      } else {
          seen[item.token_id] = i;
      }
  }
```

`arr` was edited while inside a for loop, please explain why this won't cause unexpected behaviour.

Comment (superokid) [reply packages/frontend/src/redux/slices/nft/index.js:1]: that the behavior want to achieve preserve index when adding tokens & remove duplicate
const arr = [...(n || []), ...tokens];

we can remove duplicate token using uniqBy in lodash, but that function get first unique, but the intended want to make last unique

another way maybe to make the code more obvious ```_.reverse(uniqBy(_.reverse(arr), 'token_id'))```

Comment (SteveKok) [reply packages/frontend/src/redux/slices/nft/index.js:1]: Try this:

```typescript
function keepLastUniqueDataWithProperty<
    ArrayItem extends Required<{ [key in PropertyName]: any }>,
    PropertyName extends string
>(array: ArrayItem[], propertyName: PropertyName): ArrayItem[] {
    const newArray: ArrayItem[] = []
    
    for (let i = array.length - 1; i >= 0; i--) {
        const arrayItem = array[i];

        if (newArray.filter((newArrayItem) => newArrayItem[propertyName] === arrayItem[propertyName]).length === 0) {
            newArray.unshift(arrayItem);
        }
    } 

    return newArray;
}
```

Is this what you wanted?

**Review by SteveKok — APPROVED** [review]
seems good enough

**Pull request merged**